### PR TITLE
Closes #30 | Add palito dataset

### LINKED ISSUE #30: Create dataset loader for PALITO Corpus
Dataloader name: `palito/palito.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?palito

|     Dataset| palito |
|-------------|---|
| Description | The PALITO Corpus is a compilation of multilingual written texts in four Philippine languages: Tagalog, Cebuano, Ilocano, and Hiligaynon. The categories of the corpus are mostly between literary or creative writing texts (such as novels and stories) and religious texts (such as the Bible). The overall corpus contains about 150,000 words from the literary texts and 100,000 words from the religious texts, with a total of 250,000. Previous works have used the corpus for language modeling and language identification. |
| Subsets | - |
| Languages | tgl, ceb, ilo, hil, pag, pam, war, bcl |
| Tasks | Language Modeling, Language Identification |
| License | GNU Lesser General Public License family (lgpl) |
| Homepage | https://github.com/imperialite/Philippine-Languages-Online-Corpora/tree/master/PALITO%20Corpus |
| HF URL | - |
| Paper URL | https://aclanthology.org/Y09-2024/ |

### PR DESCRIPTION
**Pull request by ilhamfp**
Closes #30 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/palito/palito.py
Comment (gentaiscool) [inline seacrowd/sea_datasets/palito/palito.py:64]: Please add Dataset in the class name (e.g., PalitoDataset).

**Review by gentaiscool — CHANGES_REQUESTED** [review]

File: seacrowd/sea_datasets/palito/palito.py
Comment (ilhamfp) [reply seacrowd/sea_datasets/palito/palito.py]: Updated 👍

File: seacrowd/sea_datasets/palito/palito.py
Comment (sabilmakbar) [inline seacrowd/sea_datasets/palito/palito.py:18]: consider changing the `_CITATION` to BibTex Format, as usually LaTeX is generally used by Academia.
You can find the BibTex format from this:
image

```suggestion
_CITATION = """
@inproceedings{dita-etal-2009-building,
    title = "Building Online Corpora of {P}hilippine Languages",
    author = "Dita, Shirley N.  and
      Roxas, Rachel Edita O.  and
      Inventado, Paul",
    editor = "Kwong, Olivia",
    booktitle = "Proceedings of the 23rd Pacific Asia Conference on Language, Information and Computation, Volume 2",
    month = dec,
    year = "2009",
    address = "Hong Kong",
    publisher = "City University of Hong Kong",
    url = "https://aclanthology.org/Y09-2024",
    pages = "646--653",
}
"""
```

**Review by sabilmakbar — CHANGES_REQUESTED** [review]
a suggested change on `_CITATION`

**Comment by sabilmakbar** [comment]
Hi @ilhamfp, mind running the code formatter first on your dataloader and ensuring no errors are thrown from there?

I found some issues when executing the formatter via
```bash
make check_file=seacrowd/sea_datasets/palito/palito.py
```

image

File: seacrowd/sea_datasets/palito/palito.py
Comment (ilhamfp) [reply seacrowd/sea_datasets/palito/palito.py:18]: Thanks!

**Comment by ilhamfp** [comment]
Formatted 👍 
image

**Comment by sabilmakbar** [comment]
Not sure why the `stale.yml` is accidentally included in this PR, mind removing it? Thanks!

**Comment by ilhamfp** [comment]
Yeah, sorry about that. Updated 👍

File: seacrowd/sea_datasets/palito/palito.py
Comment (sabilmakbar) [inline seacrowd/sea_datasets/palito/palito.py:62]: use `_LICENSE` from this:

```python
from seacrowd.utils.constants import Licenses
```

```suggestion
_LICENSE = Licenses.LGPL.value
```

**Review by sabilmakbar — CHANGES_REQUESTED** [review]
few more nits before testing the dataloader

File: seacrowd/sea_datasets/palito/palito.py
Comment (ilhamfp) [reply seacrowd/sea_datasets/palito/palito.py]: Updated 👍

File: seacrowd/sea_datasets/palito/palito.py
Comment (ilhamfp) [reply seacrowd/sea_datasets/palito/palito.py]: updated 👍

**Comment by sabilmakbar** [comment]
Hi @ilhamfp, the dataset and all of its checklist looks okay, except for one file that's misnamed on the source. I'm opening a PR to them for fixing that, then I'll approve the PR if [this PR](https://github.com/imperialite/Philippine-Languages-Online-Corpora/pull/1) has been merged to their `master`.

cc @imperialite for visibility.

**Review by sabilmakbar — COMMENTED** [review]
one more suggestion (not compulsory to implement)

File: seacrowd/sea_datasets/palito/palito.py
Comment (sabilmakbar) [inline seacrowd/sea_datasets/palito/palito.py:90]: suggestion (not compulsory to implement): use the following to avoid hardcoded val
```suggestion
    subsets = [f"{_DATASETNAME}_{lang}" for lang in _LANGUAGES]
```

**Review by sabilmakbar — COMMENTED** [review]
one more suggestion (not compulsory to implement)

File: seacrowd/sea_datasets/palito/palito.py
Comment (ilhamfp) [reply seacrowd/sea_datasets/palito/palito.py]: Updated 👍

**Review by sabilmakbar — APPROVED** [review]
LGTM! Thanks @ilhamfp for implementing this and happy holiday! Let's wait for @gentaiscool's review before merging this.

**Review by gentaiscool — APPROVED** [review]
lgtm. Thanks for the contribution.

**Pull request merged**
